psi4/src/psi4/libmints/writer.cc

```diff
@@ -469,27 +507,48 @@ void FCHKWriter::write(const std::string &filename) {
                     for (int col = 0; col < 3; ++col) transmat->set(offset + row, offset + col, pureP[row][col]);
             }
         } else {
-            // Cartesians - P orbitals are fine, but higher terms need reordering
-            if (am == 2) {
-                for (int row = 0; row < 6; ++row)
-                    for (int col = 0; col < 6; ++col) transmat->set(offset + row, offset + col, cartD[row][col]);
-            }
-            if (am == 3) {
-                for (int row = 0; row < 10; ++row)
-                    for (int col = 0; col < 10; ++col) transmat->set(offset + row, offset + col, cartF[row][col]);
-            }
-            if (am == 4) {
-                for (int row = 0; row < 15; ++row)
-                    for (int col = 0; col < 15; ++col) transmat->set(offset + row, offset + col, cartG[row][col]);
+            // Cartesians - S and P orbitals are fine, but higher terms need reordering
```

Comment:
Why change this? It does not seem that you have altered the effect of the code but to use `case` which Psi4 stays away from.

Reply:
The separate if clauses weren't aesthetic, so I wanted to use the syntax that exists in the language for this case.

Reverted to if else if.